Honestly, I don't understand how to make this task some other way, bu…

BastovOleksandr opened the pull request:
Honestly, I don't understand how to make this task some other way, but I'm sure the current solution is wrong. I need to "chain" these validators and descriptors differently.
Moreover, I have used the IntegerRange descriptor in two child classes but not in the parent one, which could be more laconic and without code repeating.
And validate() methods are the same in both child classes....

BastovOleksandr committed:
Honestly, I don't understand how to make this task some other way, but I'm sure the current solution is wrong. I need to "chain" these validators and descriptors differently.
Moreover, I have used the IntegerRange descriptor in two child classes but not in the parent one, which could be more laconic and without code repeating.
And validate() methods are the same in both child classes....

BastovOleksandr committed:
unnecessary abstract method "validate()" from SlideLimitationValidator has been replaced by a regular method with the same realization for both child classes

BastovOleksandr committed:
unnecessary "abstractmethod" import has been removed

MNDonut commented on app/main.py:
```diff
@@ -1,25 +1,75 @@
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        if not (isinstance(value, int)
+                and self.min_amount <= value <= self.max_amount):
+            value = None
```
Read this message please
https://mate.academy/chats/244801?selectedMessageId=3129519

MNDonut commented on app/main.py:
```diff
@@ -1,25 +1,75 @@
+    def __init__(
+        self,
+        name: str,
+        limitation_class: type
```
instead of type you can use SlideLimitationValidator because all your validators inherit from it

MNDonut commented on app/main.py:
```diff
@@ -1,25 +1,75 @@
+        self.height = height
+        self.weight = weight
+
+    def validate(self) -> bool:
+        if self.age and self.weight and self.height:
+            return True
+        return False
```
this validate should check if visitor's age, height and weight is correct according to age height weight of this validator

MNDonut commented on app/main.py:
```diff
@@ -1,25 +1,75 @@
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        if not (isinstance(value, int)
+                and self.min_amount <= value <= self.max_amount):
```
It should just set value. as it's integer range class, values should be integers, thus check it

BastovOleksandr committed:
I have decided to delete the unnecessary (in this case) method validate() and to use a regular attribute that shows us the actual result of the descriptors' work

also irrelevant check for INT in __set__ method of IntegerRange class has been removed into the __init__ of the SlideLimitationValidator class

oliasymak24 commented on app/main.py:
```diff
@@ -1,25 +1,75 @@
+    def __init__(
+        self,
+        name: str,
+        limitation_class: SlideLimitationValidator()
```
Do you need brackets here?

BastovOleksandr committed:
SlideLimitationValidator()    <- the are no more brackets
